consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java

```diff
@@ -69,16 +69,26 @@ public static TimestampSchedule createTimestamp(
             DEFAULT_CHAIN_ID,
             ProtocolSpecAdapters.create(
                 config.getShanghaiTime().orElse(0),
-                (specBuilder) ->
-                    MergeProtocolSchedule.applyMergeSpecificModifications(
-                        specBuilder, config.getChainId())),
+                MergeProtocolSchedule::applyMergeSpecificModificationsForShanghai),
             privacyParameters,
             isRevertReasonEnabled,
             config.isQuorum(),
             EvmConfiguration.DEFAULT)
         .createTimestampSchedule();
   }
 
+  // TODO Withdrawals remove this as part of https://github.com/hyperledger/besu/issues/4788
+  private static ProtocolSpecBuilder applyMergeSpecificModificationsForShanghai(
+      final ProtocolSpecBuilder specBuilder) {
+
+    return specBuilder
+        .blockProcessorBuilder(MergeBlockProcessor::new)
+        .blockHeaderValidatorBuilder(MergeProtocolSchedule::getBlockHeaderValidator)
+        .blockReward(Wei.ZERO)
+        .difficultyCalculator((a, b, c) -> BigInteger.ZERO)
```

Comment:
I wish Java would hurry up and support `_` as ignored lambda params.

Reply:
(__, __, __) -> ... doesn't quite have the same ring to it